docs/user-documentation/glossary.md

```diff
@@ -0,0 +1,79 @@
+ 
+## Entity 
+A Drupal term for an item of either content or configuration data, although in common usage, the term often refers to content entities. Examples include content items, custom blocks, taxonomy terms, and definitions of content types; the first three are content entities, and the last is a configuration entity. 
+ 
+## Fedora (Repository Software)
+Fedora is a digital asset management architecture upon which institutional repositories, digital archives, and digital library systems might be built. Fedora is the underlying architecture for a digital repository, and is not a complete management, indexing, discovery, and delivery application. 
+ 
+The Fedora repository functions as the standard _smart storage_ for Islandora.
+ 
+## Field
+Data of a certain type that is attached to a content entity. For instance, on a Resource Node content type, you might have fields for a title, description, display hints, subjects, and other metadata. 
+ 
+## Flysystem
+Flysystem is a filesystem abstraction library for PHP. Islandora uses Flystsem to swap about different backend filesystem applications, and comes with a custom Flystsem adapter for Fedora.
+ 
+## Ingest
+To ingest an object is to add an entry for it in Islandora. This is done primarily through the Drupal interface. 
+ 
+The word 'ingest' is used in repository language because repository software, rather than having an object and associated components forced upon it, takes in and processes the object itself, creating derivatives automatically. Often, when referring to the process of ingesting objects, Islandora and its associated documentation use other terms such as 'import' or 'add resource node'. In such contexts, these terms generally refer to the ingest process.
+ 
+## Media
+A drop-in replacement for the Drupal core upload field with a unified User Interface where editors and administrators can upload, manage, and reuse files and multimedia assets. 
+ 
+## Microservice
+A software development technique —a variant of the service-oriented architecture (SOA) structural style— that arranges an application as a collection of loosely coupled services[1] In a microservices architecture, services are fine-grained and the protocols are lightweight.
+ 
+## Open Source
+Open source describes a method of software development that promotes access to the end product's source code. Islandora is an open source product with an active development community, operating under the GPL license (2.0) for Drupal components and the MIT license for non-Drupal components. 
+ 
+## Module
+Software (usually PHP, JavaScript, and/or CSS) that extends site features and adds functionality. The Drupal project distinguishes between core and contributed modules.
```

Comment:
"Drupal modules conform to a specific structure allowing them to integrate with the Drupal architecture."